presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2409,14 +2427,26 @@ public String toString()
         private final Optional<List<String>> fileNames;
         private final PartitionStatistics statistics;
         private final PartitionStatistics statisticsUpdate;
+        private final boolean partitionActionBySync;
```

Comment:
It seems all members in PartitionAndMore are objects/nouns that describes a partition. But partitionActionBySync is a source of action. Is there another better way we can pass it in to prepareAddPartition?

Reply:
I thought about optimizing the implementation but `prepareAddPartition` is at commit phase which is too late to be aware of the state of added partition whether it's coming from the general insert path or sync utility logic.

Reply:
Thank you @yingsu00 for suggestion. It makes sense not to add a source of action in `PartitionAndMore`, the flag instead is now added to `HDFSContext` structure to indicate whether path validation is needed or not.